### PR TITLE
Initial MSC3083 + MSC3244 support

### PR DESCRIPTION
**Pull request by t3chguy**
Requires https://github.com/matrix-org/matrix-js-sdk/pull/1745
Fixes https://github.com/vector-im/element-web/issues/17686
Fixes https://github.com/vector-im/element-web/issues/17661

https://github.com/matrix-org/matrix-doc/pull/3083
https://github.com/matrix-org/matrix-doc/pull/3244

### CODE REVIEW

**Comment by t3chguy** [comment]
image
image
image
image
image
image
image
image
image
image
image
image
image

File: src/settings/handlers/RoomSettingsHandler.ts [hunk 1]
```diff
@@ -92,12 +92,12 @@ export default class RoomSettingsHandler extends MatrixClientBackedSettingsHandl
         if (settingName === "urlPreviewsEnabled") {
             const content = this.getSettings(roomId, "org.matrix.room.preview_urls") || {};
             content['disable'] = !newValue;
-            return MatrixClientPeg.get().sendStateEvent(roomId, "org.matrix.room.preview_urls", content);
+            return MatrixClientPeg.get().sendStateEvent(roomId, "org.matrix.room.preview_urls", content).then();
```
Comment (gsouquet) [inline src/settings/handlers/RoomSettingsHandler.ts:95]: What are those `then` achieving?

File: res/css/views/dialogs/_ManageRestrictedJoinRuleDialog.scss [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            display: flex;
+            margin-top: 12px;
+
+            > div {
+                flex-grow: 1;
+            }
+
+            img.mx_RoomAvatar_isSpaceRoom,
+            .mx_RoomAvatar_isSpaceRoom img {
+                border-radius: 4px;
```
Comment (gsouquet) [inline res/css/views/dialogs/_ManageRestrictedJoinRuleDialog.scss:71]: I'm wondering whether we should have a `kind` prop on the avatar component.

There seems to be three types of avatar:
- Circle
- Small border radius
- Slightly larger border radius

That would definitely reduce the local overrides for those CSS properties

**Review by gsouquet — APPROVED** [review]

File: src/settings/handlers/RoomSettingsHandler.ts (hunk 1 above)
Comment (t3chguy) [reply src/settings/handlers/RoomSettingsHandler.ts:95]: matching the signature, which is `Promise<void>` but sendStateEvent is `Promise<{ event_id: string }>`

File: res/css/views/dialogs/_ManageRestrictedJoinRuleDialog.scss (hunk 2 above)
Comment (t3chguy) [reply res/css/views/dialogs/_ManageRestrictedJoinRuleDialog.scss:71]: possibly, but would rather not grow this diff any further if possible

**Comment by niquewoodhouse** [comment]
Just some small things, mostly to do with copy:

- Change `public (anyone)` to `public` [1]
- ~~Change visible to space members to Space members can find and join~~

- Where we refer to a Space name in these dialogs, it would be good to emphasize the name with bold eg here:
image

- Is it my eyes or (in above screenshot) is 'visible to space members' here smaller than the topic and name inputs?  It might be there's a discrepancy in font sizes between figma and reality, again.  Matching the other inputs would be good if there is a discrepancy.

- Consistency in verbs: change 'view and join' to 'find and join' as we can't be sure if you can view before joining (I think)?
image

- Change below copy to be: ....'if a space is selected**,** it's members **can** ~~will be able to~~ find and join Room name'
image

- The designs had globe and lock iconography for when a room is public and private.  It was positioned in the dropdown.  We could move it to somewhere else to keep things simple, eg in the header or next to the copy underneath the dropdown - but I think it would be good to somehow suggest what those icons mean here. Where's easiest?  Doesn't have to be in the dropdown to just explain what the icon means.

[1] My only other suggestion was `Public (anyone can join)` but I just thought the copy underneath kind of explain it anyway.

**Comment by t3chguy** [comment]
image

Have tweaked all other copy as specified too

**Comment by t3chguy** [comment]
image
image

File: src/components/views/dialogs/CreateRoomDialog.tsx [hunk 3]
```diff
@@ -95,7 +107,7 @@ export default class CreateRoomDialog extends React.Component<IProps, IState> {
             createOpts.creation_content = { 'm.federate': false };
         }
 
-        if (!this.state.isPublic) {
+        if (this.state.joinRule !== JoinRule.Public) {
```
Comment (gsouquet) [inline src/components/views/dialogs/CreateRoomDialog.tsx:110]: Should this be turned in an `else` statement instead?

**Review by gsouquet — APPROVED** [review]
All the recent changes seam sensible to me 👍

**Review by niquewoodhouse — APPROVED** [review]

File: src/components/views/dialogs/CreateRoomDialog.tsx (hunk 3 above)
Comment (t3chguy) [reply src/components/views/dialogs/CreateRoomDialog.tsx:110]: oh, yeah 🤦

**Pull request merged**